packages/passport/sdk-sample-app/src/components/zkevm/EthSendTransactionExamples/NFTApproval.tsx

```diff
@@ -0,0 +1,186 @@
+
+  const { zkEvmProvider } = usePassportProvider();
+
+
+  const handleSetApproveType = useCallback((e: React.ChangeEvent<HTMLSelectElement>) => {
+    setChoosedApproveType(e.target.value as ApproveType)
+  }, [])
+
+  useEffect(() => {
+    const nftTokenId = tokenId.trim() === '' ? '1' : tokenId;
+    try {
+      const data = choosedApproveType ===ApproveType.NFTApprove
+      ? nftApproveContract.encodeFunctionData('approve', [toAddress, nftTokenId])
+      : nftApproveContract.encodeFunctionData('setApprovalForAll', [toAddress, true]);
+
+      setParams([{
+        from: fromAddress,
+        to: erc721ContractAddress,
+        value: '0',
+        data,
+      }]);
+    } catch (err) {
+      setParams([{
+        from: fromAddress,
+        to: erc721ContractAddress,
+        value: '0',
+        data: '0x',
+      }]);
+    }
+  }, [fromAddress, erc721ContractAddress, toAddress, tokenId, choosedApproveType, nftApproveContract]);
+
+  useEffect(() => {
+    const getAddress = async () => {
+      if (zkEvmProvider) {
+        const [walletAddress] = await zkEvmProvider.request({
+          method: 'eth_requestAccounts',
+        });
+        setFromAddress(walletAddress || '');
+      }
+    };
+
+    getAddress().catch(console.log);
+  }, [zkEvmProvider, setFromAddress]);
+
+  const handleSubmit = useCallback(async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    await handleExampleSubmitted({
+      method: 'eth_sendTransaction',
+      params,
+    });
+  }, [params, handleExampleSubmitted]);
+
+  return (
+    <Accordion.Item eventKey="2">
+      <Accordion.Header>NFT Approve</Accordion.Header>
+      <Accordion.Body>
+        <Form onSubmit={handleSubmit} className="mb-4">
+          <Form.Group className="mb-3">
+            <Form.Label>
+              Preview
+            </Form.Label>
+            <Form.Control
+              readOnly
+              as="textarea"
+              rows={7}
+              value={JSON.stringify(params, null, '\t')}
+              style={{
+                fontSize: '0.8rem',
+              }}
+            />
+          </Form.Group>
+          <Form.Label>Token Type</Form.Label>
+          <Form.Select onChange={handleSetApproveType}>
+            <option value={ApproveType.NFTApprove}>NFT Approve</option>
+            <option value={ApproveType.NFTApprovalForAll}>NFT Batch Approve</option>
+          </Form.Select>
+          <Form.Group className="mb-3">
+            <Form.Label>
+              From
+            </Form.Label>
+            <Form.Control
+              required
+              disabled
+              type="text"
+              placeholder="From Address"
+              value={fromAddress}
+            />
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Label>
+              {choosedApproveType === ApproveType.NFTApprove ? "ERC721" : "ERC721 / ERC1155"} Contract Address
```

Comment:
@carmen0208 there may be a few linting issues that need resolving, as `yarn workspace @imtbl/passport-sdk-sample-app build` fails

Reply:
I will check it, thanks ,you reminds me.